compiler/src/iree/compiler/Dialect/HAL/Transforms/test/BUILD.bazel

```diff
@@ -46,3 +49,33 @@ iree_lit_test_suite(
         "@llvm-project//llvm:FileCheck",
     ],
 )
+
+# Tests requiring the vulkan-spirv target backend.
+iree_cmake_extra_content(
+    content = """
+if(IREE_TARGET_BACKEND_VULKAN_SPIRV)
+""",
+    inline = True,
+)
+
+iree_lit_test_suite(
+    name = "lit_vulkan",
+    srcs = ["assign_target_devices.mlir"],
```

Comment:
This test explicitly runs a pass pipeline with `iree-hal-assign-target-devices{targets=vulkan-spirv}`. I think it's fair to have tests requiring specific backends and that the build would be the place where we make that work. I'm not familiar enough with this specific pass to tell if it really needs to be backend-specific.

Reply:
(I'm asking for the error logs. The test isn't backend-specific, it's just using two backends as written now, but we might be able to use generic names or something... need to see the failure mode to suggest more)

Reply:
https://gist.github.com/bjacob/a2b79cb5e05c57346991fe43df16a3bd

Reply:
`error: target backend 'vulkan-spirv' not registered; registered backends: llvm-cpu, vmvx, vmvx-inline`

IIRC, we assume that VMVX is always enabled and don't have any build system branches for it. Maybe the test could use `vmvx` and `vmvx-inline`?

Reply:
I've also had some ideas for a null / no-op compiler target and/or HAL driver, which could be used in tests and could be forked when creating new backends. It would probably be weird for tools to report such a target in the list of available targets though.

Reply:
I'm not familiar enough with this `iree-hal-assign-target-devices` pass to tell, but the word `devices` in it suggests that it may be interesting to keep testing with a target backend that goes with a non-`local` device? `vmvx` is like `llvm-cpu` as far as devices go.

Reply:
Sent https://github.com/openxla/iree/pull/14119. I _think_ runtime 'device' is a different enough concept from the 'target devices' being used at this compilation phase. I at least don't think it particularly matters for what this test is covering.

Reply:
That's merged, so you can sync and revert the `compiler/src/iree/compiler/Dialect/HAL/Transforms/test/` changes :D